runtime: Clean outdated vendored dependencies

LandonTClipp opened the pull request:
It was discovered that an old dependency on the CDI spec (github.com/container-orchestrated-devices/container-device-interface) was being used. This caused errors when cold-plugging devices as the CDI spec only supported cdiVersions up to 0.6.0. An example from containerd logs:

```
r="rpc error: code = Unknown desc = failed to start sandbox \"0b61ca2c130a9cbbe27b22840e954fd7436beb355081df9a495f29745dcc9290\": failed to create containerd task: failed to create shim task: device cold plug failed: cold plug: CDI device injection failed: CDI registry refresh failed: failed to load CDI Spec invalid CDI Spec: invalid version \"1.0.0\"\nfailed to load CDI Spec invalid CDI Spec: invalid version \"1.0.0\""
```

The project moved to `tags.cncf.io/container-device-interface`. We remove references to the old repo and re-do a `go mod vendor`. Re-vendoring also appears to have caught some other stale and missing vendored dependencies. So this is the reason some other projects have appeared in the `vendor/` directory.

Refs:
- https://github.com/kata-containers/kata-containers/pull/12087
- https://katacontainers.slack.com/archives/C879ACQ00/p1765838855961059?thread_ts=1757527708.685089&cid=C879ACQ00

LandonTClipp committed:
runtime: Clean outdated vendored dependencies

It was discovered that an old dependency on the CDI spec (github.com/container-orchestrated-devices/container-device-interface)
was being used. This caused errors when cold-plugging devices as the CDI spec only supported
cdiVersions up to 0.6.0. An example from containerd logs:

```
r="rpc error: code = Unknown desc = failed to start sandbox [...] failed to load CDI Spec invalid CDI Spec: invalid version \"1.0.0\""
```

The project moved to `tags.cncf.io/container-device-interface`. We remove references
to the old repo and re-do a `go mod vendor`. Re-vendoring also appears to have
caught some other stale and missing vendored dependencies. So this is the reason
some other projects have appeared in the `vendor/` directory.

Refs:
- https://github.com/kata-containers/kata-containers/pull/12087
- https://katacontainers.slack.com/archives/C879ACQ00/p1765838855961059?thread_ts=1757527708.685089&cid=C879ACQ00

Signed-off-by: LandonTClipp <11232769+LandonTClipp@users.noreply.github.com>

zvonkok reviewed (approved):
LGTM, thanks @LandonTClipp

LandonTClipp committed:
runtime: Modify injectDevices with new CDI API

The CDI API was changed slightly, so the injectDevices function must
be updated with the new API.

Signed-off-by: LandonTClipp <11232769+LandonTClipp@users.noreply.github.com>

fidencio commented:
I'm closing this one as we ended up removing all the vendored code in a different series.
